File: easy/349.intersection-of-two-arrays.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.HashSet;

/*
 * @lc app=leetcode id=349 lang=java
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        // time O(n)
        HashSet<Integer> set = new HashSet<Integer>();
        for (int num: nums1) {
            set.add(num);
        }

        HashSet<Integer> intersect = new HashSet<Integer>();
        for (int num: nums2) {
            if (set.contains(num)) {
                intersect.add(num);
            }
        }

        int[] ret = new int[intersect.size()];
        int i = 0;
        for (int num: intersect) {
            ret[i] = num;
            i++;
        }

        return ret;
    }
}
// @lc code=end

File: easy/349.intersection-of-two-arrays.js
/*
 * @lc app=leetcode id=349 lang=javascript
 *
 * [349] Intersection of Two Arrays
 * [hashmap]
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
  let set = new Set();
  nums1.forEach(num => {
    set.add(num);
  })

  let ans = new Set();
  nums2.forEach(num => {
    if (set.has(num)) {
      ans.add(num);
    }
  });

  return [...ans];
}
// @lc code=end

